### ISSUE TITLE
💅 False-positive `noUnusedVariables` with TS interface declaration merging

### ISSUE THREAD

**Issue by echocrow** [biomejs/biome#6644]
### Environment information

```bash
CLI:
  Version:                      2.0.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.7.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    correctness/noUnusedVariables

Workspace:
  Open Documents:               0
```

### Rule name

noUnusedVariables

### Playground link

https://biomejs.dev/playground/?code=aQBuAHQAZQByAGYAYQBjAGUAIABUAGgAaQBuAGcAcwAgAHsACgAgACAAZgBvAG8AOgAgAHMAdAByAGkAbgBnAAoAfQAKAAoAaQBuAHQAZQByAGYAYQBjAGUAIABUAGgAaQBuAGcAcwAgAHsACgAgACAAYgBhAHIAOgAgAHMAdAByAGkAbgBnAAoAfQAKAAoALwAvACAAVQBzAGkAbgBnACAAYABUAGgAaQBuAGcAcwBgACAAYQBjAHQAdQBhAGwAbAAgAHIAZQBmAGUAcgBlAG4AYwBlAHMAIABhAGwAbAAgAGkAbgB0AGUAcgBmAGEAYwBlACAAZABlAGYAaQBuAGkAdABpAG8AbgBzAC4ACgAvAC8AIABzAGUAZQAgAGgAdAB0AHAAcwA6AC8ALwB3AHcAdwAuAHQAeQBwAGUAcwBjAHIAaQBwAHQAbABhAG4AZwAuAG8AcgBnAC8AZABvAGMAcwAvAGgAYQBuAGQAYgBvAG8AawAvAGQAZQBjAGwAYQByAGEAdABpAG8AbgAtAG0AZQByAGcAaQBuAGcALgBoAHQAbQBsAAoAdAB5AHAAZQAgAEsAZQB5ACAAPQAgAGsAZQB5AG8AZgAgAFQAaABpAG4AZwBzAAoALwAvACAAXgA%2FACAAJwBmAG8AbwAnACAAfAAgACcAYgBhAHIAJwAgAHwAIAAnAGIAYQB6ACcACgAKAGkAbgB0AGUAcgBmAGEAYwBlACAAVABoAGkAbgBnAHMAIAB7AAoAIAAgAGIAYQB6ADoAIABzAHQAcgBpAG4AZwAKAH0ACgAKAC8ALwAgAEIAbwBnAHUAcwAgAGMAbwBkAGUAIAB0AG8AIABtAGEAcgBrACAAYABLAGUAeQBgACAAYQBzACAAdQBzAGUAZAAuAAoAZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABkAG8AUwB0AHUAZgBmACgAawBlAHkAOgAgAEsAZQB5ACkAIAB7AAoAIAAgAHIAZQB0AHUAcgBuACAAawBlAHkACgB9AAoA&language=ts

### Expected result

- **Expected:** Identically-named interfaces are all detected as used when that interface name is referenced anywhere
  -  This is because of TS's [Declaration Merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)
- **Actual:** Only the last interface definition before the reference/usage is detected as used

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
@Conaclos do we handle interface merging inside your semantic model?

**Comment by Conaclos** [comment]
No it is not handled.
